Update tooling

rmarren1 opened the pull request:
Update tooling like in https://github.com/plotly/dash-core-components/pull/299

Also, this moved `requirements.txt -> .circleci/dev/dev-requirements.txt` and makes that file minimal, like in this PR: https://github.com/plotly/dash/pull/439

rmarren1 commented:
Should there be a demo for this repo?
`npm run start` does not work.

`webpack.server.config.js` points to `./src/demo/index.js`, but that file does not exist.

T4rk1n commented on .circleci/config.yml:
```diff
@@ -24,21 +24,29 @@ jobs:
               - "node_modules"
 
+      - run:
+          name: prettier --list-different
```
Why prettier ? It's auto generated code.

rmarren1 replied:
True, this would probably just get annoying. I'll remove it

T4rk1n commented on scripts/publish.js:
```diff
@@ -0,0 +1,28 @@
+        throw new Error('\nIt looks like there are uncommitted changes! Aborting until these changes have been resolved.\n');
+    } else {
+        execSh([
+                'npm publish --otp',
```
Think you can remove the --otp, it will asks if need be.

rmarren1 replied:
Okay, wasn't sure what that did, I just use `npm publish` then allow it to `401` and ask for the OTP. This was copied from the `dash-core-components` repo, so this happens there as well.

rmarren1 commented:
@T4rk1n care to take another look?

The CI error in python3.7 is a E104 so it should fix itself with another run; however, it doesn't look like I have permissions to re-trigger builds for this repo (I do for other ones though). @chriddyp  can I get permissions to do that in the future?

T4rk1n commented on package.json:
```diff
@@ -20,14 +20,20 @@
-    "start": "webpack-serve ./webpack.serve.config.js --open",
-    "test": "eslint --fix src",
     "uninstall-local": "pip uninstall dash-html-components -y",
+    "prepublish": "npm run clean && npm run generate-components && npm test && npm run build:js && npm run build:js-dev && npm run build:py",
```
Can we condense this scripts by using `build:all` ?

rmarren1 commented:
@T4rk1n 
- Made that simplification to prepublish
- Removed some more scripts that are now obselete (after removing archetype)
- Updated README with more up to data instructions
- Re-built the whole thing -- for some reason this created a bunch of diffs, but I am not sure what changed. It looks like that issue with `^M`, but I thought that was fixed

rmarren1 commented:
@T4rk1n (reminder)

rmarren1 commented:
publish script is same as in DCC, so that problem will exist there too.

rmarren1 commented:
Not sure the best way to solve this, some options I can think of:
* Write out own script in a higher level language (e.g. python) that takes care of compatibility
* have a `clean` and `clean:windows` command simultaneously

T4rk1n commented on package.json:
```diff
@@ -13,21 +13,19 @@
-    "test": "eslint --fix src",
+    "install-local": "python setup.py install",
     "uninstall-local": "pip uninstall dash-html-components -y",
+    "prepublish": "npm run clean && npm run generate-components && npm run build:all",
```
We should remove prepublish entirely and use the new npm methods it warns about instead. This will throw when you install on windows.

T4rk1n commented:
Let's not fix those platform issues for now, clean/generate will run under cygwin/gitbash.

Windows dev can just run the commands to publish separately, that's what I do.

rmarren1 commented:
:+1:
